tests/test_sparse_gp.cc

```diff
@@ -204,4 +204,108 @@ TYPED_TEST(SparseGaussianProcessTest, test_likelihood) {
   EXPECT_NEAR(expected, actual, 1e-2);
 }
 
+struct FixedInducingPoints {
+
+  FixedInducingPoints(double min, double max, std::size_t num_points = 10)
+      : min_(min), max_(max), num_points_(num_points) {}
+
+  template <typename CovarianceFunction>
+  std::vector<double> operator()(const CovarianceFunction &cov,
+                                 const std::vector<double> &features) const {
+    return linspace(min_, max_, num_points_);
+  }
+
+  double min_;
+  double max_;
+  std::size_t num_points_;
+};
+
+TEST(test_sparse_gp, test_update_exists) {
+
+  using CovFunc = SquaredExponential<EuclideanDistance>;
+  using MeanFunc = ZeroMean;
+  using GrouperFunction = LeaveOneIntervalOut;
+  using InducingPointStrategy = UniformlySpacedInducingPoints;
+
+  using SparseGPR =
+      SparseGaussianProcessRegression<CovFunc, MeanFunc, GrouperFunction,
+                                      InducingPointStrategy>;
+
+  using FitType = Fit<SparseGPFit<double>>;
+
+  EXPECT_TRUE(bool(has_valid_update<SparseGPR, FitType, double>::value));
+  EXPECT_TRUE(bool(can_update_in_place<SparseGPR, FitType, double>::value));
+}
+
+TYPED_TEST(SparseGaussianProcessTest, test_update) {
+  auto grouper = this->grouper;
+  auto covariance = make_simple_covariance_function();
+  auto dataset = make_toy_linear_data();
+
+  const double min =
+      *std::min_element(dataset.features.begin(), dataset.features.end());
+  const double max =
+      *std::max_element(dataset.features.begin(), dataset.features.end());
+
+  FixedInducingPoints strategy(min, max, 8);
+  auto sparse =
+      sparse_gp_from_covariance(covariance, grouper, strategy, "sparse");
+  sparse.set_param(details::inducing_nugget_name(), 1e-3);
+  sparse.set_param(details::measurement_nugget_name(), 1e-12);
+
+  auto groups = dataset.group_by(grouper).groups();
+  const std::size_t original_size = groups.size();
```

Comment:
`original_size` is clear in context, but could be confused with `dataset.size()` instead of `num_groups`.

Reply:
done